### ISSUE TITLE
[🚀 Feature]: JSpecify Nullness annotations for Java

### ISSUE THREAD

**Issue by mk868** [SeleniumHQ/selenium#14291]
### Feature and motivation

I suggest adding JSpecify Nullness annotations to the Selenium framework code.
These annotations allow you to specify which parameters and return values can be null.
I'm aware that information about potential null values are already placed in the JavaDoc, but using annotations will be transparent to IDEs and static code analyzers.
This will give developers better exposure of potential problems with their code in order to avoid NullPointerExceptions.

Using annotations will also improve interoperability with Kotlin.

Links:
- https://jspecify.dev/
- https://github.com/jspecify/jspecify

### Usage example

The annotated method would look like this:
```java
// ...

import org.jspecify.annotations.Nullable;

public interface WebElement extends SearchContext, TakesScreenshot {
// ...

@Nullable String getAttribute(String name);

// ...
}
```

Then the IDE/static code analyzer will display a warning about unsafe code:
```java
var element = webDriver.findElement(By.tagName("img"));
var cssClasses = element.getAttribute("class").split(" "); // warning here, possible NullPointerException
```

**Comment by mk868** [comment]
I can contribute to add these annotations.
Can I just create a Pull Request with changes?
Or Is additional analysis/confirmation needed for this feature by the Selenium team?

**Comment by diemol** [comment]
@mk868, yes, please. Focus on small PRs so they can be quickly reviewed and merged. It does not matter if you send many PRs. A PR with 2-3 files changed will be merged rather quickly, but a PR with 20 files changed will take several days.

**Comment by mk868** [comment]
The task is still valid

**Comment by iampopovich** [comment]
@diemol Since the task is closed as completed:
Will newly added Nullable annotations still be considered in PRs?

I noticed internal null checks in class methods, but class fields lack Nullable markers. While not critical, adding annotations would better enforce class contracts.
(Not insisting if changes are unwarranted.)

**Comment by diemol** [comment]
It got closed automatically after merging the PR.

**Comment by iampopovich** [comment]
@diemol, sorry for a couple of similar PRs. They contain a lot of boilerplate for adding annotations to exception classes. For such trivial changes, is it better to submit them as several small pull requests or one large one?

**Comment by iampopovich** [comment]
There are quite a lot of classes left without annotations. Should I send larger batches of changes for review? Or are changes of 1-2 classes acceptable for a quick review?

**Comment by diemol** [comment]
1-2 classes is better

**Comment by mk868** [comment]
This isn't fully completed (it was probably closed automatically by GitHub), can we reopen this issue?

**Comment by mk868** [comment]
My idea is to perform a full NullAway analysis on the project (only production code, not test code).  
We currently have the NullAway plugin configured, but we haven't plugged it into CI/CD yet.  
To enable NullAway in CI/CD, we need to fix all existing NullAway errors detected in the code.


Personally, I run a local NullAway analysis (script below) to find these errors, and then I create related PRs with fixes.

```
#!/bin/bash

export JAVA_HOME="$HOME"/.jdks/azul-22.0.2
export PATH="$JAVA_HOME/bin:$PATH"

bazel clean
bazel build //java/src/... \
  --//java:nullaway_level=ERROR \
  --javacopt=-XDaddTypeAnnotationsToSymbol=true \
  --java_runtime_version=local_jdk \
  --tool_java_runtime_version=local_jdk \
  --extra_toolchains=@local_jdk//:all
```

**Comment by iampopovich** [comment]
I have a flight now. I'll be at the computer this evening and use the script. Thanks 🙏

**Comment by iampopovich** [comment]
We worked on a very similar problem with ruff errors for the Python client with Corey Goldberg for about a couple of weeks. I think it can be done fairly quickly here too.

**Comment by asolntsev** [comment]
What I am strongly suggesting is:
Let's mark with `@NullsMarked` the whole package (in file "packafge-info.java") instead of polluting every single class, field, method and parameter. 

Let everything be non-nullable by default,  and mark with `@Nullable` only the specific places.

**Comment by asolntsev** [comment]
In the end, this request was gradually implemented by many people in many pull requests:

https://github.com/SeleniumHQ/selenium/pulls?q=is%3Apr+is%3Aclosed+%2314291+

Big thanks to
* @mk868 
* @iampopovich 
* @spencerarq 
* @asolntsev
